Automattic/jetpack
# Charts Component: Add Charts Tooltip Component

## Proposed changes:

This PR introduces a new tooltip component 

**Changes Made:**

**Dependencies:**
- Added @visx/event and @visx/tooltip to dependencies.

**Exports:**
- Updated index.ts to export the new Tooltip component.
- BarChart Component:
- Integrated tooltip functionality to display data points on hover.
- Added showTooltips prop to enable/disable tooltips.

**Storybook:**
- Adds a new storybook + stories to demonstrate usage of tooltips
- Added a new story to demonstrate the BarChart with tooltips.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* In your Jetpack local env, navigate to the Storybook development environment:

   ```
  pnpm i
  cd projects/js-packages/storybook
  pnpm run storybook:dev
  ```

* Find `JS Packages/Charts/Tooltip` in the sidebar

* Verify the tooltips chart:
  - Renders correctly with sample data
  - Responds to different data sets via Storybook controls

* Find `JS Packages/Charts/Bar Chart` in the sidebar

* navigate to the `With Tooltips` story
  - check the bar chart renders with tooltips on mouse over

![image](https://github.com/user-attachments/assets/41131426-5de6-4c3b-bd5c-04391c490c68)
